src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -39,6 +30,72 @@ def child_sort_key(item):
     return [item["start_timestamp"], item["timestamp"]]
 
 
+def query_trace_data(trace_id, params):
+    transaction_query = discover.prepare_discover_query(
+        selected_columns=[
+            "id",
+            "transaction.status",
+            "transaction.op",
+            "transaction.duration",
+            "transaction",
+            "timestamp",
+            # project gets the slug, and project.id gets added automatically
+            "project",
+            "trace.span",
+            "trace.parent_span",
+            'to_other(trace.parent_span, "", 0, 1) AS root',
+        ],
+        # We want to guarantee at least getting the root, and hopefully events near it with timestamp
+        # id is just for consistent results
+        orderby=["-root", "-timestamp", "id"],
+        params=params,
+        query=f"event.type:transaction trace:{trace_id}",
+    )
+    error_query = discover.prepare_discover_query(
+        selected_columns=[
+            "id",
+            "project",
+            "timestamp",
+            "trace.span",
+            "transaction",
+            "issue",
+            "title",
+            "tags[level]",
+        ],
+        # Don't add timestamp to this orderby as snuba will have to split the time range up and make multiple queries
+        orderby=["id"],
+        params=params,
+        query=f"!event.type:transaction trace:{trace_id}",
+        auto_fields=False,
+    )
+    snuba_params = [
+        SnubaQueryParams(
+            dataset=Dataset.Discover,
+            start=snuba_filter.start,
+            end=snuba_filter.end,
+            groupby=snuba_filter.groupby,
+            conditions=snuba_filter.conditions,
+            filter_keys=snuba_filter.filter_keys,
+            aggregations=snuba_filter.aggregations,
+            selected_columns=snuba_filter.selected_columns,
+            having=snuba_filter.having,
+            orderby=snuba_filter.orderby,
+            limit=MAX_TRACE_SIZE,
+        )
+        for dataset, snuba_filter in [transaction_query.filter, error_query.filter]
```

Comment:
Test is breaking `'Filter' object is not iterable`

Reply:
woops ty, fixed in 9ce1fcba54a57365eca0dc4c6df804546923e9f3